### PR TITLE
Support installation on Apple Silicon

### LINKED ISSUE #5306: Add support for Apple M1 ?
Appreciate your work, thank you.

I am unable to install on Apple M1 using Eclipse 4.20M3 for Arm64/AArch64 :

```
Cannot complete the install because some dependencies are not satisfiable
  org.eclipse.equinox.p2.iu; net.resheim.eclipse.launcher.feature.group [3.0.2.202004261020,3.0.2.202004261020] cannot be installed in this environment because its filter is not applicable.
```

Maybe the x86_64 feature filter needs to be removed ? Just guessing.

Thanks again.

### PR DESCRIPTION
**Pull request by kwin**
Lift restrictions on architecture and window system for the feature
This closes #5306

### CODE REVIEW

File: net.resheim.eclipse.launcher-feature/feature.xml [hunk 1]
```diff
@@ -5,9 +5,7 @@
       version="3.0.2.qualifier"
       provider-name="%providerName"
       plugin="net.resheim.eclipse.launcher.core"
-      os="macosx"
-      ws="cocoa"
-      arch="x86_64">
+      os="macosx">
```
Comment (guw) [inline net.resheim.eclipse.launcher-feature/feature.xml:8]: Depending on how this is built the Tycho target platform configuration also needs an adjustment.

**Comment by turesheim** [comment]
Thanks for the patch! Need to replace Travis-CI with GitHub Actions, but there will be a new build soon.

**Comment by turesheim** [comment]
Builds now look OK, please test installing from https://github.com/turesheim/eclipse-launcher/suites/5687919717/artifacts/187033191

**Comment by guw** [comment]
Different error now
```
Cannot complete the install because one or more required items could not be found.
  Software being installed: Eclipse Launcher Utility for macOS 3.0.3.202203161944 (net.resheim.eclipse.launcher.feature.group 3.0.3.202203161944)
  Missing requirement: Eclipse Launcher Utility for macOS 3.0.3.202203161944 (net.resheim.eclipse.launcher.feature.group 3.0.3.202203161944) requires 'org.eclipse.equinox.p2.iu; net.resheim.eclipse.launcher.macosx [3.0.3.202203161944,3.0.3.202203161944]' but it could not be found
  ```

**Pull request merged**